### DISCUSSION TITLE
Looking for meeting type CHIP

### DISCUSSION THREAD

**Discussion by mpalsf** [code4recovery/12-step-meeting-list#1421] [General]
I don't see Chip (for chip meeting) as a default meeting type. Am I missing it?

**Comment by joshreisner** [comment]
hi @mpalsf at some point `Chips` was renamed to `Birthday` by request, however the code is still `H` from that era. 

we recommend using the Birthday type for this, but you can rename it locally on your site if you prefer: https://github.com/code4recovery/tsml-ui?tab=readme-ov-file#add-custom-types

all the types: https://github.com/code4recovery/spec?tab=readme-ov-file#meeting-types

**Comment by mpalsf** [comment]
Great Josh, thanks!

**Comment by mpalsf** [comment]
![types](https://github.com/code4recovery/12-step-meeting-list/assets/30127450/116a15fa-115f-4d7b-93fb-a3d42e5b3e47)

Here's a screen shot of our default list. It doesn't include Birthday.

**Reply by kiyote33** [reply]
Hi @mpalsf, 
When editing any meeting, an administrator can add types to the list of available meeting types by clicking on the View all link. If you check the Birthdays checkbox, it will become available to all meetings in the future. Hope this helps...
![image](https://github.com/code4recovery/12-step-meeting-list/assets/57125427/8e5d0b1f-7160-47f9-89fe-ddb0d211ab84)

**Reply by joshreisner** [reply]
good advice for TSML. however since aasfmarin is on airtable you're going to want to add some Birthday meetings there somehow and get that type showing up in the TSML view

**Reply by mpalsf** [reply]
OK, so different backend because of Airtable. I do see a Focus and a Format field in Airtable. They each have hardcoded multiselect list that seems to just have the label and not a place to use a code (H) and a label. If we don't have the unique code, how are we populating meeting types when this goes to Meeting Guide?

**Reply by joshreisner** [reply]
it's been a while, but i think that the airtable importer i made for yall expects to see the long name of the meeting type, and that it handles translating the types into codes

looking here, i see three "Chip" types that it is expecting to translate from AASF: https://github.com/code4recovery/sheets/blob/main/app/Http/Controllers/AirtableController.php#L102-L104

do you see any "chip" meetings currently in airtable? if so, maybe that code isn't working anymore

**Reply by joshreisner** [reply]
[this is the airtable view](https://airtable.com/appdNCw2EOfkca0bz/tblQ7auDpQRZYH6OH/viw2Ci0NkQDHgL73z?blocks=hide) the importer sees, i don't think any chip or birthday types are in the types column

**Reply by mpalsf** [reply]
I'm getting very close. I've worked through the various clean up and transformation happening in Airtable between the Meetings table and the final TSML table in Airtable. The Chip Meeting item is in the TSML table in the Type field. (see Too Early). 

1) When I click Manual Sync in the TSML view, there are about 300 errors returned that Online is not defined as a Type. So, where does Code for Recovery want the Online value. Are these errors fatal to the sync?

2) The changes shown in the TSML table are not pushing to Meeting Guide when I click Manual Sync. For testing I put an "x" on the end of Too Early (Too Earlyx) and that meeting includes Chip Meeting in type. They are both in the TSML sync table.

3) When I clicked on https://airtable-json.aasfmarin.org/refresh in our local documentation, I get a Secure Connection Failed error, code: SSL_ERROR_NO_CYPHER_OVERLAP. Is that killing our sync?

We definitely had some holes in our local documentation, so I am updating that as I go for future reference.

thanks.

**Reply by joshreisner** [reply]
1. that's completely fine - not fatal to the sync. we don't have a type for `Online`, we just infer it from the presence of a valid `conference_url` and/or `conference_phone`
2. the app takes a ~12 hours to sync up - i would first just test that your changes are present on your meetings page
3. you should update the documentation to point here: https://sheets.code4recovery.org/aasfmarin - airtable-json.aasfmarin is obsolete and i think might mean that there's a service out there that could be canceled or removed

🎉 i see a birthday type on your list now: https://aasfmarin.org/meetings?type=birthday

<img width="292" alt="Screenshot 2024-05-21 at 12 09 44 PM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/1f94bd0c-ffdf-4806-8896-50032f75f7b1">

so you should be seeing that in meeting guide now / soon

if you want that to read `Chips` on your list you can update your existing TSML UI customization in wordpress to look like this:

```js
var tsml_react_config = {
  timezone: "America/Los_Angeles",
  conference_providers: {
    "sites.google.com": "Custom",
    "tinyurl.com": "Custom",
    "MillValleyCabin.com": "Custom",
    "drydocksf.org": "Custom",
  },
  show: {
    cityAsRegionFallback: false,
  },
  strings: {
    en: {
      types: {
        H: "Chips",
      },
    },
  },
};
```

**Reply by mpalsf** [reply]
Well, eureka, it is showing up now!

One last one: where does the var tsml_react_config = {}?

I see add_shortcode() in tsml_react.php that references the conference providers, but the syntax is a little different.
============================================
add_shortcode('tsml_react', function() {
    $mapbox_key = 'pk.eyJ1IjoiYWFzZi1tcGFsIiwiYSI6ImNqaHl6NHY0NzByemszcHBkcHpib296OHkifQ.F7GfpF1jRosyQ0PGNUfDkA';
    $conference_providers = [
        'sites.google.com' => 'Custom',
        'tinyurl.com' => 'Custom',
        'MillValleyCabin.com' => 'Custom',
        'drydocksf.org' => 'Custom',
    ];
    $is_local = substr(get_site_url(), -5) == '.test';
    $host = 'https://tsml-ui.' . ($is_local ? 'test' : 'code4recovery.org');
    # $url = 'https://airtable-json.' . ($is_local ? 'test' : 'aasfmarin.org');
    $url = 'https://sheets.code4recovery.org/' . ($is_local ? 'test' : 'storage/aasfmarin.json');
    wp_enqueue_style('tsml_react', $host . '/style.css');
    wp_enqueue_script('tsml_react', $host . '/app.js');
    wp_localize_script('tsml_react', 'tsml_react_config', array(
        'timezone' => get_option('timezone_string'),
        'conference_providers' => $conference_providers,
        'show' => array(
            'cityAsRegionFallback' => false,
        )
    ));
    return '<div id="tsml-ui" 
		data-src="' . $url . '" 
		data-mapbox="' . $mapbox_key . '"
		data-timezone="America/Los_Angeles"></div>';
});

**Reply by joshreisner** [reply]
oh yeah, i was thinking it was in the frontend but looks like it's in this custom shortcode.

i would add this inside, right after the `cityAsregionFallback` line

```php
  'strings' => array(
    'en' => array(
      'types' => array(
        'H' => 'Chips',
      ),
    ),
  ),
```

**Reply by mpalsf** [reply]
OK, got it. Does that kick in during the TSML export or locally in the Meeting Guide? Should it be immediate?

**Reply by joshreisner** [reply]
it should be immediate on your site, it won't affect Meeting Guide. those meetings will show up as `Birthday` unless there is a change by GSO

**Reply by mpalsf** [reply]
OK. It still says Birthday on the web site: https://aasfmarin.org/find-a-meeting?type=birthday

**Reply by mpalsf** [reply]
add_shortcode('tsml_react', function() {
    $mapbox_key = 'pk.eyJ1IjoiYWFzZi1tcGFsIiwiYSI6ImNqaHl6NHY0NzByemszcHBkcHpib296OHkifQ.F7GfpF1jRosyQ0PGNUfDkA';
    $conference_providers = [
        'sites.google.com' => 'Custom',
        'tinyurl.com' => 'Custom',
        'MillValleyCabin.com' => 'Custom',
        'drydocksf.org' => 'Custom',
    ];
    $is_local = substr(get_site_url(), -5) == '.test';
    $host = 'https://tsml-ui.' . ($is_local ? 'test' : 'code4recovery.org');
    # $url = 'https://airtable-json.' . ($is_local ? 'test' : 'aasfmarin.org');
    $url = 'https://sheets.code4recovery.org/' . ($is_local ? 'test' : 'storage/aasfmarin.json');
    wp_enqueue_style('tsml_react', $host . '/style.css');
    wp_enqueue_script('tsml_react', $host . '/app.js');
    wp_localize_script('tsml_react', 'tsml_react_config', array(
        'timezone' => get_option('timezone_string'),
        'conference_providers' => $conference_providers,
        'show' => array(
            'cityAsRegionFallback' => false,
		'strings' => array(
    		'en' => array(
      		'types' => array(
        		'H' => 'Chip Meeting',
      		),
    		),
  			),
        )
    ));
    return '<div id="tsml-ui" 
		data-src="' . $url . '" 
		data-mapbox="' . $mapbox_key . '"
		data-timezone="America/Los_Angeles"></div>';
});

**Reply by joshreisner** [reply]
<img width="482" alt="Screenshot 2024-05-22 at 10 24 23 AM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/e5666803-810c-4cd9-b6c2-11b28a3ef516">

whoops, move the block i suggested down a level, so that `strings` is at the same level as `show`. i think you can basically take one of the closing parens and move it above `strings`

**Reply by mpalsf** [reply]
Hey, you're right again!! That was fun. Thanks

**Reply by joshreisner** [reply]
Glad to see it's working now! If it were me I'd take the word "Meeting" out - similar to how `Online` or `Outdoor` don't have the word "Meeting" in there

<img width="915" alt="Screenshot 2024-05-22 at 11 02 12 AM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/cf44ee3c-68db-4e28-9089-5203bd43ac74">

**Reply by mpalsf** [reply]
Right, thanks.
